java/challenges/lib/src/main/java/challenges/blog/InsertionSort.java

```diff
@@ -0,0 +1,21 @@
+package challenges.blog;
+
+public class InsertionSort {
+
+public static int[] Insertion(int[] arr){
```

Comment:
Even though you stated you were unsure of how it works, looks like you have a working insertion sort coded out!